src/ui/page/share/dataset/steps/metadataFilling/metadataFilling.tsx

```diff
@@ -0,0 +1,346 @@
+            <p>{t('share.metadataFilling.topic')}</p>
+            <Field
+              id={id7}
+              label={t('share.metadataFilling.topicSelection')}
+              onChange={handleFieldValueChange(id7)}
+              value={formItemFieldValue(id7)}
+            />
+          </div>
+        ),
+        style: {
+          side: 'right'
+        }
+      },
+      {
+        id: id8,
+        type: 'text',
+        label: 'geographicalCoverage',
+        value: [],
+        render: (): JSX.Element => (
+          <div className="okp4-dataverse-portal-share-data-metadata-filling">
+            <p>{t('share.metadataFilling.geographicalCoverage')}</p>
+            <Field
+              id={id8}
+              label={t('share.metadataFilling.geographicalCoverageSelection')}
+              onChange={handleFieldValueChange(id8)}
+              value={formItemFieldValue(id8)}
+            />
+          </div>
+        ),
+        style: {
+          side: 'right'
+        }
+      },
+      {
+        id: id9,
+        type: 'text',
+        label: 'tags',
+        value: [],
+        render: (): JSX.Element => (
+          <div className="okp4-dataverse-portal-share-data-metadata-filling">
+            <p>{t('share.metadataFilling.tags')}</p>
+            <Field
+              id={id9}
+              label={'tags'}
+              onChange={handleFieldValueChange(id9)}
+              value={formItemFieldValue(id9)}
+            />
+          </div>
+        ),
+        style: {
+          side: 'right'
+        }
+      },
+      {
+        id: id10,
+        type: 'text',
+        label: 'fee',
+        value: 0,
+        render: (): JSX.Element => (
+          <div className="okp4-dataverse-portal-share-data-metadata-filling fee">
+            <Field
+              id={id10}
+              label={'fee'}
+              onChange={handleFieldValueChange(id10)}
+              value={formItemFieldValue(id10)}
+            />
+          </div>
+        ),
+        style: {
+          side: 'right'
+        }
+      }
+    ]
+  }, [formItemFieldValue, handleFieldValueChange, t])
+
+  const mapForm = (form: DatasetForm): initFormPayload => {
+    return form.map(formItem => ({
+      id: formItem.id,
+      type: formItem.type,
+      label: formItem.label,
+      value: formItem.value,
+      required: formItem.required
+    }))
+  }
+
+  const handleFormError = useCallback(
+    (error: ResourceError) => {
+      const { title, type } = formErrorData(error)
+      console.error(ShowFileError.show(error))
+      dispatchNotification({
+        type: type,
+        titleKey: title
+      })
+    },
+    [dispatchNotification]
+  )
+  useEffect(() => {
+    if (!isFormInitialized()()) {
+      const initialForm = mapForm(datasetForm)
+      pipe(initialForm, initForm, IOE.mapLeft(handleFormError))()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [datasetForm, initForm, isFormInitialized])
```

Comment:
@0xLodz 
I removed `handleFormError` explicitly because it causes multiples unmount each time a value changes in any field component. (probably due to curryfication)
We must investigate but for now it's the best solution to remove it from the dependencies array.